pkg/apiserver/logsearch/task.go

```diff
@@ -74,7 +74,8 @@ func (tg *TaskGroup) SyncRun() {
 
 	// Create log directory
 	dir := path.Join(tg.service.logStoreDirectory, strconv.Itoa(int(tg.model.ID)))
-	if err := os.MkdirAll(dir, 0777); err == nil {
+	err := os.MkdirAll(dir, 0777) // #nosec
```

Comment:
Why do we need a nosec for this checked error?

Reply:
There's a least privilege issue: `Expect directory permissions to be 0750 or less`. I'm not sure about the perm in this case.